### PR TITLE
fix: invoke origin get_list, if call isn't gp specific

### PR DESCRIPTION
**Pull request by blaggacao**
this ensures that the upstream business logic is not bypassed

Prior to this Fix:
```javascript
		frappe.call({
			method: "frappe.client.get_list",
			args: {
				doctype: "UOM Conversion Factor",
				filters: { category: __("Length") },
				fields: ["to_uom"],
				limit_page_length: 500,
			},
			callback: function (r) {
				r.message.forEach((row) => units.push(row.to_uom));
			},
		});
```

```sql
SELECT `["to_uom"]` FROM `tabUOM Conversion Factor` WHERE `name`='{"category":"Length"}' LIMIT 20
```

Note how `["to_uom"]` is not a valid column name and `20` is not the right limit.

### CODE REVIEW

File: gameplan/extends/client.py [hunk 1]
```diff
@@ -5,21 +5,36 @@
...
+	controller = get_controller(doctype)
+	if hasattr(controller, "get_list_query"):
+		check_permissions(doctype, kwargs.get("parent"))
+		query = frappe.qb.get_query(
+			table=doctype,
+			fields=kwargs.get("fields"),
+			filters=kwargs.get("filters"),
+			order_by=kwargs.get("order_by"),
+			offset=kwargs.get("start"),
+			limit=kwargs.get("limit", 20),
```
Comment (blaggacao) [inline gameplan/extends/client.py:19]: Note how the signature differs from `frappe` framework. But remediating this seems to be a more involved refactoring.

File: gameplan/extends/client.py [hunk 2]
```diff
@@ -5,21 +5,36 @@
...
+		if return_value is not None:
+			query = return_value
+		return query.run(as_dict=True, debug=kwargs.get("debug"))
+	return frappe.client.get_list(
+		doctype,
+		fields=kwargs.get("fields"),
+		filters=kwargs.get("filters"),
+		order_by=kwargs.get("order_by"),
+		limit_start=kwargs.get("limit_start"),
+		limit_page_length=kwargs.get("limit_page_length",20),
```
Comment (blaggacao) [inline gameplan/extends/client.py:32]: ... `frappe` uses different kwarg names.

File: gameplan/extends/client.py [hunk 3]
```diff
@@ -5,21 +5,36 @@
...
-		table=doctype,
-		fields=fields,
-		filters=filters,
-		order_by=order_by,
-		offset=start,
-		limit=limit,
-		group_by=group_by,
+def get_list(doctype, **kwargs):
+	controller = get_controller(doctype)
+	if hasattr(controller, "get_list_query"):
```
Comment (blaggacao) [inline gameplan/extends/client.py:11]: A future refactoring may consider to trigger this code path exclusively on gameplan models. Without e.g. namespacing `get_list_query` or additional gp-specific signal attributes, this logic has too a high probability of one day causing unintended side effects.

File: gameplan/extends/client.py [hunk 4]
```diff
@@ -5,21 +5,36 @@
...
+		check_permissions(doctype, kwargs.get("parent"))
```
Comment (blaggacao) [inline gameplan/extends/client.py:12]: I hope scoping this logic to what is supposed to be a gp-specific controller, only is the right thing to do an gp didn't depend on generally modified check_permissions. :pray: 

If that's the case, it makes things a bit more difficult of course. In that case, all gp requests could send a sentinel value to help trigger (or not) that logic.

File: gameplan/extends/client.py (hunk 3 above)
Comment (blaggacao) [reply gameplan/extends/client.py:11]: I see that all doctypes are namespaced with `gp_` maybe that can be used upfront as a cheap string-prefix check. But these type of improvements is not part of this fix.

**Comment by blaggacao** [comment]
@netchampfaris Have you had a chance to look at this? Not sure what the side effect exactly is, but given that upstream is also a moving target, it's probably best to reduce exposure to api-incompatibility as much as possible as done in this PR.

**Review by netchampfaris — COMMENTED** [review]
@blaggacao Hey, thanks for your concern. However, this get_list function is only a stop-gap solution until Frappe Framework's `get_list` starts using `frappe.qb.get_query` under the hood. So far, I don't think GP's get_list should be any different from a standard get_list call.

**Comment by blaggacao** [comment]
Understood, I got construed a `["to_uom"]` vs expected `"to_uom"`. It may be that this is then a hidden bug in frappe upstream and will surface once frappe's `get_list` would use `frappe.qb.get_query`, as well.

It might be worthwhile to keep this PR open as a hint for people encountering the same issue and seeking for a temporary patch.

**Comment by ankush** [comment]
This needs to removed, the fix is only temporary... run GP on separate site if it's a real problem for you (for now)

**Comment by blaggacao** [comment]
Since I somehow lost track of this and I'm updating my stack nowis.

> the fix is only temporary..

Is this already fixed or do I still need to patch?

**Comment by blaggacao** [comment]
@netchampfaris although this fix was closed, it appears that it still cleanly applies on latest `main` indicating that it is still necessary (or otherwise there's still a silent bug in the code). How do we proceed?

**Comment by ankush** [comment]
This was fixed long ago https://github.com/frappe/gameplan/commit/9cf205543f629fece4ad3698c414b936395e147f

**Comment by blaggacao** [comment]
Ah nice! thanks for the headsup! :handshake:

**Pull request closed without merging**